exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebServer.java

```diff
@@ -408,5 +448,47 @@ public void close() throws Exception {
     if (embeddedJetty != null) {
       embeddedJetty.stop();
     }
+    //Deleting temp directory
+    FileUtils.deleteDirectory(getTmpJavaScriptDir());
+  }
+
+  /* Static wrapper text for JavaScript content */
+  final String FILE_CONTENT_HEADER =
+      "  //Returns description\n" +
+      "  function getDescription(optionName) {\n" +
+      "    if (!(optionName in optsDescripMap)) {\n" +
+      "      return \"\";\n" +
+      "    }\n" +
+      "    return optsDescripMap[optionName];\n" +
+      "  }\n" +
+      "  var optsDescripMap = {";
+  final String FILE_CONTENT_FOOTER =
+      "};\n";
+
+  //Generate Options Description JavaScript
+  private void generateOptionsDescriptionJSFile() throws IOException {
+    //Obtain properties from Resource file
+    Properties optionsDescriptions = new Properties();
+    InputStream resrcInputStream = this.getClass().getClassLoader().getResourceAsStream(OPTIONS_DESCRIBE_PROPERTIES);
+    optionsDescriptions.load(resrcInputStream);
+    int numLeftToWrite = optionsDescriptions.size();
+
+    File optionsDescriptionFile = new File(getTmpJavaScriptDir(), OPTIONS_DESCRIBE_JS);
+    optionsDescriptionFile.deleteOnExit();
+    logger.info("Will write {} descriptions to {}", numLeftToWrite, optionsDescriptionFile.getAbsolutePath());
+    BufferedWriter writer = new BufferedWriter(new FileWriter(optionsDescriptionFile));
```

Comment:
Is try with resources possible here?

Reply:
SHould be. I'll incorporate that change.